protocols/virtualfund/virtualfund-single-hop_test.go

```diff
@@ -97,58 +77,56 @@ func TestSingleHopVirtualFund(t *testing.T) {
 		TurnNum: 0,
 		IsFinal: false,
 	}
-
-	var AsAlice = func(t *testing.T) {
-
-		/////////////////////
-		// BEGIN test data //
-		/////////////////////
-
-		// In this test, we play Alice
-		my := alice
-
-		// Alice plays role 0 so has no ledger channel on her left
-		var ledgerChannelToMyLeft *channel.TwoPartyLedger
-
-		ledgerManager := ledger.NewLedgerManager()
-		left := outcome.Allocation{Destination: alice.destination, Amount: big.NewInt(5)}
-		right := outcome.Allocation{Destination: p1.destination, Amount: big.NewInt(5)}
-		myIndex := uint(0) // because Alice is in the "left" slot
-
-		// She has a single ledger channel L_0 connecting her to P_1
-		var ledgerChannelToMyRight, _ = ledger.CreateTestLedger(left, right, &alice.privateKey, myIndex, big.NewInt(0))
-		// Ensure this channel is fully funded on chain
-		ledgerChannelToMyRight.OnChainFunding = ledgerChannelToMyRight.PreFundState().Outcome.TotalAllocated()
-
-		///////////////////
-		// END test data //
-		///////////////////
-
-		testNew := func(t *testing.T) {
-
-			// Assert that a valid set of constructor args does not result in an error
-			o, err := New(false, vPreFund, my.address, n, my.role, ledgerChannelToMyLeft, ledgerChannelToMyRight)
-			if err != nil {
-				t.Error(err)
-			}
-
-			got := o.ToMyRight.ExpectedGuarantees[types.Address{}] // VState only has one (native) asset represented by the zero address
-			var expectedGuaranteeMetadata = outcome.GuaranteeMetadata{Left: ledgerChannelToMyRight.MyDestination(), Right: ledgerChannelToMyRight.TheirDestination()}
-			var expectedEncodedGuaranteeMetadata, _ = expectedGuaranteeMetadata.Encode()
-			var expectedGuarantee outcome.Allocation = outcome.Allocation{
-				Destination:    o.V.Id,
-				Amount:         big.NewInt(0).Set(vPreFund.VariablePart().Outcome[0].TotalAllocated()),
-				AllocationType: outcome.GuaranteeAllocationType,
-				Metadata:       expectedEncodedGuaranteeMetadata,
+	vPostFund := vPreFund.Clone()
+	vPostFund.TurnNum = 1
+
+	TestAs := func(my actor, t *testing.T) {
+
+		// IMPORTANT these are templates. Clone them before using to prevent sharing mutable data between tests
```

Comment:
We could move creating these ledger channels into a function and just call that function from each test. This would avoid having to remember to clone the channels at the start of the test.

Reply:
7b98783e540dabd6fc6bee914bbb55909a1708ae